core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3503,10 +3504,13 @@ private boolean isLocalScaledWriterExchange(PlanNode node)
             Optional<PlanNode> result = searchFrom(node)
                     .where(planNode -> planNode instanceof ExchangeNode && ((ExchangeNode) planNode).getScope() == LOCAL)
                     .findFirst();
-
-            return result.isPresent()
-                    && result.get() instanceof ExchangeNode
-                    && ((ExchangeNode) result.get()).getPartitioningScheme().getPartitioning().getHandle().equals(SCALED_WRITER_DISTRIBUTION);
+            if (result.isPresent() && result.get() instanceof ExchangeNode exchangeNode) {
+                PartitioningHandle handle = exchangeNode.getPartitioningScheme().getPartitioning().getHandle();
+                return handle.equals(SCALED_WRITER_DISTRIBUTION)
+                        || handle.equals(SCALED_WRITER_HASH_DISTRIBUTION)
+                        || handle.getConnectorHandle() instanceof ScaleWriterPartitioningHandle;
+            }
```

Comment:
nit: we might want to keep scaling (enabled) info in `TableWriterNode`. I'm don't have strong opinion here though as I don't know how it would look like finally